relayer/src/worker/connection.rs

```diff
@@ -0,0 +1,127 @@
+use std::{thread, time::Duration};
+
+use anomaly::BoxError;
+use crossbeam_channel::Receiver;
+use tracing::{debug, warn};
+
+use crate::connection::Connection as RelayConnection;
+use crate::telemetry::Telemetry;
+use crate::{
+    chain::handle::ChainHandlePair, object::Connection, util::retry::retry_with_index,
+    worker::retry_strategy,
+};
+
+use super::WorkerCmd;
+
+pub struct ConnectionWorker {
+    connection: Connection,
+    chains: ChainHandlePair,
+    cmd_rx: Receiver<WorkerCmd>,
+
+    #[allow(dead_code)]
+    telemetry: Telemetry,
+}
+
+impl ConnectionWorker {
+    pub fn new(
+        connection: Connection,
+        chains: ChainHandlePair,
+        cmd_rx: Receiver<WorkerCmd>,
+        telemetry: Telemetry,
+    ) -> Self {
+        Self {
+            connection,
+            chains,
+            cmd_rx,
+            telemetry,
+        }
+    }
+
+    /// Run the event loop for events associated with a [`Connection`].
+    pub(crate) fn run(self) -> Result<(), BoxError> {
```

Comment:
Can you add some documentation to explain a bit what is the expected execution flow of the function overall? My understanding is as follows:

- Wait for incoming block events for the connection state on the source chain `a_chain` to be updated.
- Match the new state machine and update to the next connection state.
  - Only cares about the direction from `a_chain` to `b_chain`, i.e. `(State::Init, State::TryOpen)` is not handled.
- If the connection state has reached `(State::Open, State::Open)`, loop forever (should have been return right?)
- If any error happens, retry until the connection succeed? 
   - The retry is in two stages: retry a number of times for each block event update, and retry again in the next block update.